tests/subgraph/mapping-v3-core/mapping-v3-engine-flex-core.test.ts

```diff
@@ -0,0 +1,975 @@
+  clearStore();
+  // Add project to store
+  const projectId = BigInt.fromI32(0);
+  mockCoreType(TEST_CONTRACT_ADDRESS, `${coreType}`);
+  const fullProjectId = generateContractSpecificId(
+    TEST_CONTRACT_ADDRESS,
+    projectId
+  );
+  addNewProjectToStore(
+    TEST_CONTRACT_ADDRESS,
+    projectId,
+    "Test Project",
+    randomAddressGenerator.generateRandomAddress(),
+    BigInt.fromI64(i64(1e18)),
+    CURRENT_BLOCK_TIMESTAMP.minus(BigInt.fromI32(100))
+  );
+
+  const event: ExternalAssetDependencyUpdated = changetype<
+    ExternalAssetDependencyUpdated
+  >(newMockEvent());
+  event.address = TEST_CONTRACT_ADDRESS;
+  event.block.timestamp = CURRENT_BLOCK_TIMESTAMP;
+
+  const _index0 = BigInt.zero();
+  const _dependencyType0 = BigInt.zero();
+  const _externalAssetDependencyCount0 = BigInt.fromI32(1);
+
+  assert.fieldEquals(
+    PROJECT_ENTITY_TYPE,
+    fullProjectId,
+    "externalAssetDependencyCount",
+    BigInt.fromI32(0).toString()
+  );
+
+  event.parameters = [
+    new ethereum.EventParam(
+      "_projectId",
+      ethereum.Value.fromUnsignedBigInt(projectId)
+    ),
+    new ethereum.EventParam(
+      "_index",
+      ethereum.Value.fromUnsignedBigInt(_index0)
+    ),
+    new ethereum.EventParam("_cid", ethereum.Value.fromString(IPFS_CID)),
+    new ethereum.EventParam(
+      "_dependencyType",
+      ethereum.Value.fromUnsignedBigInt(_dependencyType0)
+    ),
+    new ethereum.EventParam(
+      "_externalAssetDependencyCount",
+      ethereum.Value.fromUnsignedBigInt(_externalAssetDependencyCount0)
+    )
+  ];
+
+  // add event
+  handleExternalAssetDependencyUpdated(event);
+  // checks project external asset dependency count
+  assert.fieldEquals(
+    PROJECT_ENTITY_TYPE,
+    fullProjectId,
+    "externalAssetDependencyCount",
+    _externalAssetDependencyCount0.toString()
+  );
+  // checks project's updatedAt
+  assert.fieldEquals(
+    PROJECT_ENTITY_TYPE,
+    fullProjectId,
+    "updatedAt",
+    CURRENT_BLOCK_TIMESTAMP.toString()
+  );
+  // checks project external asset dependency cid
+  assert.fieldEquals(
+    PROJECT_EXTERNAL_ASSET_DEPENDENCY_ENTITY_TYPE,
+    fullProjectId + "-" + _index0.toString(),
+    "cid",
+    IPFS_CID
+  );
+  // checks project external asset dependency dependency type
+  assert.fieldEquals(
+    PROJECT_EXTERNAL_ASSET_DEPENDENCY_ENTITY_TYPE,
+    fullProjectId + "-" + _index0.toString(),
+    "dependencyType",
+    FLEX_CONTRACT_EXTERNAL_ASSET_DEP_TYPES[_dependencyType0.toI32()]
+  );
+  // checks project external asset dependency project relationship
+  assert.fieldEquals(
+    PROJECT_EXTERNAL_ASSET_DEPENDENCY_ENTITY_TYPE,
+    fullProjectId + "-" + _index0.toString(),
+    "project",
+    fullProjectId
+  );
+  
+  const updateEvent: ExternalAssetDependencyUpdated = changetype<
+    ExternalAssetDependencyUpdated
+  >(newMockEvent());
+  updateEvent.address = TEST_CONTRACT_ADDRESS;
+  updateEvent.block.timestamp = CURRENT_BLOCK_TIMESTAMP;
```

Comment:
It would be helpful to choose another later timestamp so that we can assert that it has changed.